impeller/renderer/backend/vulkan/pipeline_vk.cc

```diff
@@ -355,19 +356,36 @@ std::unique_ptr<PipelineVK> PipelineVK::Create(
   //----------------------------------------------------------------------------
   /// Pipeline Layout a.k.a the descriptor sets and uniforms.
   ///
-  std::vector<vk::DescriptorSetLayoutBinding> desc_bindings;
+  std::vector<vk::DescriptorSetLayoutBinding> set_bindings;
+
+  vk::Sampler vk_immutable_sampler =
+      immutable_sampler ? immutable_sampler->GetSampler() : VK_NULL_HANDLE;
 
   for (auto layout : desc.GetVertexDescriptor()->GetDescriptorSetLayouts()) {
-    auto vk_desc_layout = ToVKDescriptorSetLayoutBinding(layout);
-    desc_bindings.push_back(vk_desc_layout);
+    vk::DescriptorSetLayoutBinding set_binding;
+    set_binding.binding = layout.binding;
+    set_binding.descriptorCount = 1u;
+    set_binding.descriptorType = ToVKDescriptorType(layout.descriptor_type);
+    set_binding.stageFlags = ToVkShaderStage(layout.shader_stage);
+    // TODO(csg): This specifies the immutable sampler for all sampled images.
```

Comment:
I don't think we have any cases of this today, since we should only support binding external textures in the texture_fill or tiled_texture_fill shader?

Reply:
No, we definitely do not. Which is why I didn't go down the route of handling those cases. I think the fix would be to make the `ImmutablSamplerKey` takes an ordered (by binding) set of a pair of sampler-conversion objects and use that for hash-and-cache.

Filed https://github.com/flutter/flutter/issues/143719 as a followup for this TODO. We can likely bury the priority for a long long time.